Foxtrot/Driver/Rewriting/Utils/ParameterExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Compiler;
+using System.Diagnostics.Contracts;
+
+namespace Microsoft.Contracts.Foxtrot.Utils
+{
+    internal static class ParameterExtensions
+    {
+        /// <summary>
+        /// Returns underlying parameter type.
+        /// For generic types it would be like "Func`2".
+        /// </summary>
+        public static string GetGenericTypeName(this Parameter parameter)
```

Comment:
:bulb: I would rather this be an extension method on `TypeNode`, and more accurately reflect the behavior:

```csharp
/// <summary>
/// Gets the metadata name of the type. For generic types the arity is appended, such
/// as <c>Func`2</c> for <see cref="Func{T, TResult}"/>.
/// </summary>
public static string GetMetadataName(this TypeNode typeNode)
{
  Contract.Requires(typeNode != null);
  return HelperMethods.Unspecialize(typeNode).Name.Name;
}
```